core/inlooptool.py

```diff
@@ -1003,26 +1788,215 @@ def add_build_year_to_surface(self, file_path, field_name="bouwjaar"):
+        # Get source layer definition
+        src_defn = src_layer.GetLayerDefn()
+        
+        # Get the names of the fields in the destination layer
+        dst_defn = dst_layer.GetLayerDefn()
+        dst_field_names = [dst_defn.GetFieldDefn(i).GetName() for i in range(dst_defn.GetFieldCount())]
+        
+        # Iterate through the features in the source layer
+        for src_feat in src_layer:
+            # Create a new feature in the destination layer
+            dst_feat = ogr.Feature(dst_defn)
+            
+            # Copy the fields from the source to the destination if the field names match
+            for i in range(src_defn.GetFieldCount()):
+                field_name = src_defn.GetFieldDefn(i).GetName()
+                if field_name in dst_field_names:
+                    dst_feat.SetField(field_name, src_feat.GetField(i))
+            
+            # Set the primary key manually if needed
+            if primary_key_field:
+                dst_feat.SetField(primary_key_field, src_feat.GetFID())
+            
+            # Set the geometry
+            geom = src_feat.GetGeometryRef()
+            if geom:
+                dst_feat.SetGeometry(geom.Clone())
+            
+            # Add the feature to the destination layer
+            dst_layer.CreateFeature(dst_feat)
+            
+            # Destroy the feature to free resources
+            dst_feat = None
+        
+        # Sync the data to disk
+        dst_layer.SyncToDisk()
+    
+    def _save_to_gpkg(self,file_folder,template_gpkg):
+        print("Preparing template gpkg")
+        file_name = self.set_output_name(file_folder)
+        file_path = os.path.join(file_folder, file_name)
+        self.copy_and_rename_file(template_gpkg, file_path)
+        
+        print("Saving layers to gpkg")
+        # Initialize layers with common elements
+        layers = [
+            (CHECKS_TABLE_NAME, "2_Controles"),
+            (PIPES_TABLE_NAME, "3_GWSW_leidingen"),
+            (RESULT_TABLE_NAME, "4_BGT_inlooptabel"),
+            (SURFACES_TABLE_NAME, "5_BGT_oppervlakken"),
+            (STATISTICS_TABLE_NAME, "6_Statistieken"),
+            (SETTINGS_TABLE_NAME, "7_Rekeninstellingen")
+        ]
+        
+        # Prepend the additional element if the layer exists
+        if self.mem_database.GetLayerByName(INF_PAVEMENT_TABLE_NAME_PREV) is not None:
+            layers.insert(0, (INF_PAVEMENT_TABLE_NAME_PREV, "1_Waterpasserende_verharding_en_groene_daken"))
+        
+        with self.open_gpkg(file_path) as dst_gpkg:
+            for db_layer, gpkg_layer in layers:
+                print(f"Saving {gpkg_layer} layer in gpkg")
+                self._write_to_disk(dst_gpkg, db_layer, gpkg_layer)
+                if db_layer == RESULT_TABLE_NAME:
+                    self.track_changes(dst_gpkg)
+           
+        print("All layers saved successfully.")
```

Comment:
Kan hier `return str(file_path)` aan toe worden gevoegd? VOor QGIS maakt het niets uit maar in AGP moet ik de gpkg nog naar een .gdb omzetten. Hiervoor moet ik weten wat de naam van het gpkg bestand is geworden.